vcd/resource_vcd_provider_vdc.go

```diff
@@ -0,0 +1,688 @@
+					Description: fmt.Sprintf("Used %s in this Provider VDC", typeOfCapacity),
+				},
+			},
+		},
+	}
+}
+
+// resourceVcdProviderVdc defines the resource for a Provider VDC.
+func resourceVcdProviderVdc() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdProviderVdcCreate,
+		ReadContext:   resourceVcdProviderVdcRead,
+		UpdateContext: resourceVcdProviderVdcUpdate,
+		DeleteContext: resourceVcdProviderVdcDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceProviderVdcImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the Provider VDC",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional description of the Provider VDC",
+			},
+			"status": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Status of the Provider VDC, it can be -1 (creation failed), 0 (not ready), 1 (ready), 2 (unknown) or 3 (unrecognized)",
+			},
+			"is_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "True if this Provider VDC is enabled and can provide resources to organization VDCs. A Provider VDC is always enabled on creation",
+			},
+			"capabilities": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed:    true,
+				Description: "Set of virtual hardware versions supported by this Provider VDC",
+			},
+			"compute_capacity": {
+				Type:     schema.TypeList,
+				Computed: true,
+				// MaxItems: 1 - A computed field can't use "MaxItems", this is a reminder that this is a single-element list.
+				Description: "Single-element list with an indicator of CPU and memory capacity",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cpu":    providerVdcRootCapacityUsage("CPU"),
+						"memory": providerVdcRootCapacityUsage("Memory"),
+						"is_elastic": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "True if compute capacity can grow or shrink based on demand",
+						},
+						"is_ha": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "True if compute capacity is highly available",
+						},
+					},
+				},
+			},
+			"compute_provider_scope": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Represents the compute fault domain for this Provider VDC. This value is a tenant-facing tag that is shown to tenants when viewing fault domains of the child Organization VDCs (for example, a VDC Group)",
+			},
+			"highest_supported_hardware_version": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The highest virtual hardware version supported by this Provider VDC",
+			},
+			"nsxt_manager_id": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
Missing `ForceNew: true` (I don't see `update` capability implemented)?

Reply:
Added

Reply:
On second thought, the UI allows creating a provider VDC without network pool. I don't know how good such a provider VDC would be, but perhaps we should allow it.
What do you think?

Reply:
I guess it is fine for now, don't see a lot of use cases (at least now) for a VDC without network pool. We could improve if there is ever such need